src/getLocale.ts

```diff
@@ -0,0 +1,39 @@
+import { isString } from './isString';
+import { storage } from './storage';
+import type { CountryCode } from './types/countries';
+
+const KEY = 'gu.geolocation';
+const URL = 'https://api.nextgen.guardianapps.co.uk/geolocation';
+
+// best guess that we have a valid code, without actually shipping the entire list
+const isValidCountryCode = (country: unknown) =>
+	isString(country) && /^[A-Z]{2}$/.test(country as string);
```

Comment:
Could we use the `@guardian/types`—`Country` for this?

Why do we have the whole list in this repo now?

Reply:
ooooh didn't know it was in there – totally

Reply:
Whoops, I somehow thought we’d put it in there. Ignore this for now. However my comment about having it in this repo still stands. We could test whether it is a `typeof Country`, maybe?

Reply:
this is not checking against the type though, just an approximate test (is a 2 char uppercase string)

Reply:
Gotcha `isTwoCharacterString` 😉

Reply:
précisément